traffic_ops/traffic_ops_golang/server/put_status.go

```diff
@@ -121,7 +121,7 @@ func UpdateStatusHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if *status.Name == tc.CacheStatusAdminDown.String() || *status.Name == tc.CacheStatusOffline.String() {
+	if *status.Name != tc.CacheStatusOnline.String() && *status.Name != tc.CacheStatusReported.String() {
 		if reqObj.OfflineReason == nil {
```

Comment:
this actually makes offline reason required for a bunch of statuses  so if this is a breaking api change, i will just hold off until api 5.x.